### PR TITLE
Fix dashboard routing

### PR DESCRIPTION
**Pull request by ericpgreen2**
Before this PR: navigating to a dashboard with an invalid config would result in a 500 error.

After this PR: navigating to a dashboard with an invalid config redirects to the dashboard configuration page.

### CODE REVIEW

**Comment by magorlick** [comment]
if I create an empty yaml file in the dashboard folder I can't edit it in the UI and get a 404.
image

**Comment by ericpgreen2** [comment]
> if I create an empty yaml file in the dashboard folder I can't edit it in the UI and get a 404. image

@magorlick, are you wanting the application to reactively update to new files you put in your Rill project folder? I thought that was slated for `0.17.0`. Or are you referring to something else?

Regardless, I just pushed some edits to the `/edit` page handling, which may have helped.

File: web-local/src/routes/(application)/dashboard/[name]/+page.ts [hunk 1]
```diff
@@ -10,28 +12,26 @@ export async function load({ params }) {
...
-      return {
-        metricViewName: params.name,
-      };
-    }
   } catch (err) {
-    if (
-      ExplorerMetricsDefinitionDoesntExist.includes(err.message) ||
-      err.message.includes(ExplorerMetricsDefinitionDoesntExist)
-    ) {
+    if (err.response?.data?.message.includes("entry not found")) {
```
Comment (hamilton) [inline web-local/src/routes/(application)/dashboard/[name]/+page.ts:20]: do we get anything like an enum for this? Feels a little weird that we have these hard-coded.

**Review by hamilton — APPROVED** [review]
The code looks good! 

In terms of experience, wondering if we can get a few changes in. if I click the plus on the nav to create a new dashboard, I'm greeted with a nice red error message that says `metrics view source not selected`. I feel like I'm being yelled at before I've done anything wrong. Could we (1) change this error message to something like `select a model to get started` + `select a timestamp column` for the next one, and (2) not make it red? I feel there's a difference between "this is an error" and "let's guide the user along".

I recognize we're trying to get this thing over the finish line so I defer to @magorlick on whether these should be P0 or P1.

File: web-local/src/routes/(application)/dashboard/[name]/+page.ts (hunk 1 above)
Comment (ericpgreen2) [reply web-local/src/routes/(application)/dashboard/[name]/+page.ts]: Good callout – no enum from the backend, but I created a `lib/errors/messages.ts` to quarantine the hard-coded text

**Pull request merged**